core/shared/src/main/scala/scalaz/zio/delay/Delay.scala

```diff
@@ -0,0 +1,38 @@
+package scalaz.zio.delay
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+import scalaz.zio.ZIO
+import scalaz.zio.clock.Clock
+import scalaz.zio.clock._
+import scalaz.zio.delay.Delay._
+import scalaz.zio.duration.Duration
+
+sealed trait Delay extends Serializable with Product { self =>
+  def run: ZIO[Clock, Nothing, Duration] = currentTime(unit = TimeUnit.MILLISECONDS).flatMap { millis =>
+    self match {
+      case Relative(duration)     => ZIO.succeed(duration)
+      case Absolute(instant)      => ZIO.succeed(Duration(instant.toEpochMilli - millis, TimeUnit.MILLISECONDS))
+      case Min(l, r)              => l.run.zip(r.run).map { case (d1, d2) => if (d1 < d2) d1 else d2 }
+      case Max(l, r)              => Min(r, l).run
+      case Sum(l, r)              => l.run.zip(r.run).map { case (d1, d2) => d1 + d2 }
+      case TimesFactor(l, factor) => l.run.map(d => d * factor)
+    }
+  }
+
+  def *(factor: Double) = TimesFactor(self, factor)
+  def +(delay: Delay)   = Sum(self, delay)
```

Comment:
Excellent work, this is looking really great! A few minor suggestions:

1. Rename `TimesFactor` to `Scale`
2. Add `min`, `max`